apps/src/code-studio/header.js

```diff
@@ -73,6 +74,10 @@ header.build = function(
   scriptNameData,
   hasAppOptions
 ) {
+  const store = getStore();
+  if (progressData) {
+    store.dispatch(useDbProgress());
```

Comment:
why are we doing this here _and_ in `loadApp`?

Reply:
We prefer to do this here, as it is the first point where we load progress from anywhere (DB or clientState). However, certain scripts (meaning courses/units) do not report progress as part of creating the header. For these, we instead have to wait for the call in loadApp, which happens later, to get the DB progress.

Reply:
👍